### PR TITLE
test: enable float/double/binary array casts to string

### PR DESCRIPTION
**Pull request by manuzhang**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion-comet/issues/4248

## Rationale for this change

`CometCastSuite` had FloatType, DoubleType, and BinaryType commented out for `ArrayType` to `StringType` casts. The native cast path already handles floating-point elements, and BinaryType can use the existing native binary string formatting path when arrays cast their elements to strings.

## What changes are included in this PR?

- Enable FloatType, DoubleType, and BinaryType in `CometCastSuite`.`cast ArrayType to StringType`.
- Remove the special incompatibility guard for `ArrayType(BinaryType) -> StringType` so it follows normal element string-cast support.

## How are these changes tested?

- `./mvnw test -Pspark-4.1 -Dtest=none -Dsuites="org.apache.comet.CometCastSuite cast ArrayType to StringType"`
- Also verified FloatType, DoubleType, and BinaryType one at a time with the same focused Spark 4.1 suite before restoring the full type list.

---

Co-authored-by: @codex

### CODE REVIEW

**Comment by manuzhang** [comment]
@coderfender @comphead please help review. Thanks!

**Review by andygrove — COMMENTED** [review]
Thanks @manuzhang. Changes look reasonable. Can you confirm that there are existing tests covering these changes?

**Comment by manuzhang** [comment]
@andygrove These are tests I've done locally

> * ./mvnw test -Pspark-4.1 -Dtest=none -Dsuites="org.apache.comet.CometCastSuite cast ArrayType to StringType"
> * Also verified FloatType, DoubleType, and BinaryType one at a time with the same focused Spark 4.1 suite before restoring the full type list.

**Review by coderfender — COMMENTED** [review]
Thank you for this PR @manuzhang. The only suggestion is to probably use this PR to add some more test cases to help hash out any edge cases which arent exposed in the current framework for Float, Binary and Double type ?

File: spark/src/main/scala/org/apache/comet/expressions/CometCast.scala
Comment (andygrove) [inline spark/src/main/scala/org/apache/comet/expressions/CometCast.scala]: Does this example need updating?

**Review by comphead — COMMENTED** [review]
Thanks @manuzhang should we have also `sql` tests reflecting the correct behavior?

**Comment by comphead** [comment]
this is annoying  
```
Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Restarted CI

File: spark/src/main/scala/org/apache/comet/expressions/CometCast.scala [hunk 1]
```diff
@@ -243,11 +241,7 @@ object CometCast extends CometExpressionSerde[Cast] with CometExprShim {
       case DataTypes.DateType => Compatible()
       case DataTypes.TimestampType => Compatible()
       case DataTypes.FloatType | DataTypes.DoubleType =>
-        Compatible(
-          Some(
-            "There can be differences in precision. " +
-              "For example, the input \"1.4E-45\" will produce 1.0E-45 " +
-              "instead of 1.4E-45"))
+        Compatible(Some("There can be differences in precision"))
```
Comment (comphead) [inline spark/src/main/scala/org/apache/comet/expressions/CometCast.scala]: would be still nice to give the scope of the problem to the end user

**Comment by comphead** [comment]
Thanks @manuzhang the PR looks solid to me, some minor note though and CI pending

**Review by comphead — APPROVED** [review]
Thanks @manuzhang
